internal/expr/parser.go

```diff
@@ -206,17 +219,38 @@ loop:
 		switch p.input[p.pos-1] {
 		// If the preceding byte is one of these then we might be at the start
 		// of an expression.
-		case ' ', '\t', '\n', '\r', '=', ',', '(', '[', '>', '<', '+', '-', '*', '/', '|', '%':
+		case ' ', '\t', '\n', '\r', '=', ',', '[', '>', '<', '+', '-', '*', '/', '|', '%':
+			break loop
+		case '(':
+			p.bracketLevel++
+			break loop
+		case ')':
+			p.bracketLevel--
 			break loop
 		}
 	}
 
-	p.skipBlanks()
-
+	p.partStart = p.pos
 	return nil
 
 }
 
+// startSubpart is called to allow parsing of nested parts.
+func (p *Parser) startSubpart() *checkpoint {
+	scp := p.save()
+	p.init(p.input[p.pos:])
+	return scp
+}
+
+// collectSubpart must be called to collect all parts parsed since startSubpart and move the parser up a nesting level.
+func (p *Parser) collectSubpart(scp *checkpoint) *ParsedExpr {
```

Comment:
As with `startSubpart`, I think the language can be cleaned slightly here, perhaps something like this?
```go
// collectSubpart returns the nested expression parsed since the
// corresponding sub-part checkpoint and restores the parser to
// its new position.
```

Reply:
Cleaned up.